From 89c5cc4676fb37888c5685deabd06af88deff532 Mon Sep 17 00:00:00 2001
From: Ria Thomas <ria.thomas@morsemicro.com>
Date: Mon, 10 Mar 2025 18:24:07 +1100
Subject: [PATCH] mac80211: IBSS bridge support

With the addition of BATMAN routing protocol, an IBSS (ad hoc) interface
can be bridged with Ethernet intefaces. Add bridge mode support for IBSS
interfaces.

Morse Micro note: This commit is required if IBSS is used.
---
 net/wireless/core.c | 4 ++--
 net/wireless/util.c | 1 -
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/net/wireless/core.c b/net/wireless/core.c
index 800fbfd59194..b4b154ebce9d 100644
--- a/net/wireless/core.c
+++ b/net/wireless/core.c
@@ -1404,8 +1404,8 @@ void cfg80211_init_wdev(struct wireless_dev *wdev)
 	wdev->ps_timeout = -1;
 
 	if ((wdev->iftype == NL80211_IFTYPE_STATION ||
-	     wdev->iftype == NL80211_IFTYPE_P2P_CLIENT ||
-	     wdev->iftype == NL80211_IFTYPE_ADHOC) && !wdev->use_4addr)
+	     wdev->iftype == NL80211_IFTYPE_P2P_CLIENT) &&
+        !wdev->use_4addr)
 		wdev->netdev->priv_flags |= IFF_DONT_BRIDGE;
 
 	INIT_WORK(&wdev->disconnect_wk, cfg80211_autodisconnect_wk);
diff --git a/net/wireless/util.c b/net/wireless/util.c
index 640ec502e82d..2c26d01618b6 100644
--- a/net/wireless/util.c
+++ b/net/wireless/util.c
@@ -1232,7 +1232,6 @@ int cfg80211_change_iface(struct cfg80211_registered_device *rdev,
 			fallthrough;
 		case NL80211_IFTYPE_OCB:
 		case NL80211_IFTYPE_P2P_CLIENT:
-		case NL80211_IFTYPE_ADHOC:
 			dev->priv_flags |= IFF_DONT_BRIDGE;
 			break;
 		case NL80211_IFTYPE_P2P_GO:
